src/python/pants/engine/rules.py

```diff
@@ -51,7 +52,12 @@
 @memoized
 def SubsystemRule(subsystem: Type[Subsystem]) -> TaskRule:
     """Returns a TaskRule that constructs an instance of the subsystem."""
-    return TaskRule(**subsystem.signature())
+    warn_or_error(
+        removal_version="2.17.0dev0",
+        entity=f"using `SubsystemRule({subsystem.__name__})`",
+        hint=f"Use `*{subsystem.__name__}.rules()` instead.",
+    )
+    return next(iter(subsystem.rules()))  # type: ignore[call-arg]  # mypy dislikes memoziedclassmethod
```

Comment:
Is this future proof for your upcoming changes?

Reply:
Yes.

Reply:
The overall change is future-proofing for the upcoming changes -- a new API would need to be added anyway -- so it makes sense to deprecate the existing thing.

Reply:
Pardon - I specifically meant the `next(iter())`. Will that still work when n > 1?

Reply:
@Eric-Arellano It will not return the environment-sensitive options, it can only return the main `Subsystem`. Currently published subsystems shouldn't run into this issue anyway, but we might need a backfill API in the meantime